testsuite/cluster-test/src/experiments/versioning_test.rs

```diff
@@ -0,0 +1,263 @@
+            account_1.sequence_number,
+            123456,
+            0,
+            LBR_NAME.to_owned(),
+            10,
+        )
+        .map_err(|e| format_err!("Failed to create signed transaction: {}", e))?;
+
+        account_1.sequence_number += 1;
+
+        execute_and_wait_transactions(&mut full_node_client, &mut account_1, vec![txn1.clone()])
+            .await?;
+
+        info!("3. Change the rest of the images in the second batch");
+        update_batch_instance(context, &self.second_batch, self.updated_image_tag.clone()).await?;
+
+        info!("4. Send a transaction to make sure this feature is still not activated.");
+        let txn2 = create_user_txn(
+            &account_1.key_pair,
+            txn_payload.clone(),
+            account_1.address,
+            account_1.sequence_number,
+            123456,
+            0,
+            LBR_NAME.to_owned(),
+            10,
+        )
+        .map_err(|e| format_err!("Failed to create signed transaction: {}", e))?;
+
+        account_1.sequence_number += 1;
+        execute_and_wait_transactions(&mut full_node_client, &mut account_1, vec![txn2.clone()])
+            .await?;
+
+        info!("5. Send a transaction to activate such feature");
+        let mut faucet_account = context.tx_emitter.load_faucet_account(&full_node).await?;
+        let update_txn = create_user_txn(
+            &faucet_account.key_pair,
+            TransactionPayload::Script(encode_update_libra_version_script(LibraVersion {
+                major: 11,
+            })),
+            faucet_account.address,
+            faucet_account.sequence_number,
+            123456,
+            0,
+            LBR_NAME.to_owned(),
+            10,
+        )
+        .map_err(|e| format_err!("Failed to create signed transaction: {}", e))?;
+        faucet_account.sequence_number += 1;
+
+        execute_and_wait_transactions(
+            &mut full_node_client,
+            &mut faucet_account,
+            vec![update_txn.clone()],
+        )
+        .await?;
+
+        info!("6. Send a transaction to make sure it passes the full node mempool but will not be committed by updated validators.");
+        let txn3 = create_user_txn(
+            &account_1.key_pair,
+            txn_payload,
+            account_1.address,
+            account_1.sequence_number,
+            123456,
+            0,
+            LBR_NAME.to_owned(),
+            10,
+        )
+        .map_err(|e| format_err!("Failed to create signed transaction: {}", e))?;
+
+        full_node_client
+            .submit_transaction(txn3.clone())
+            .await
+            .map_err(|e| format_err!("Transaction should pass the full node mempool: {}", e))?;
+
+        if execute_and_wait_transactions(&mut full_node_client, &mut account_1, vec![txn3.clone()])
+            .await
+            .is_ok()
+        {
+            return Err(format_err!(
+                "Transaction should not be committed by the validators"
```

Comment:
Need to propagate root cause(see comment below)

Reply:
Same here as well.